src/main/java/com/fcgl/madrid/search/service/UserSearchService.java

```diff
@@ -54,9 +60,23 @@ public UserSearchService(UserSearchRepository userSearchRepo) {
 
     }
 
-    public ResponseEntity<Response<SearchHistoryResponse>> searchByUserId(UserId userId) {
-        List<UserSearch> searchHistory = userSearchRepo.findByUserId(userId);
-        Response response = new Response<SearchHistoryResponse>(InternalStatus.OK, new SearchHistoryResponse(searchHistory));
+    /**
+     * This method returns queries in a paginated format made by a user.
+     *
+     * @param id       User id
+     * @param pageSize Number of entries per page
+     * @param page     Which page number
+     * @return response
+     */
+    public ResponseEntity<Response<SearchHistoryResponse>> searchByUserIdRecent(UserId id, int pageSize, int page) {
+        // Paginate List
+        Pageable pageableRequest = PageRequest.of(page, pageSize);
+        List<UserSearch> searchHistory = userSearchRepo.findByUserIdOrderByAddedOnDesc(id, pageableRequest);
+        // Get total number of records
+        int records = userSearchRepo.countByUserId(id);
+        // Format Response
+        PaginatedResponse paginatedResponse = new PaginatedResponse<>(new SearchHistoryResponse(searchHistory), pageSize, page + 1, records); // to user we want to show non-zero index
```

Comment:
Oh I see. The +1 is for the user. In the app you scroll down to go to the next page. We're using 0 in all the other services though. Maybe the frontend should be in charge of this logic if needed?

Reply:
I agree, made that change to reflect this.